### PR TITLE
[feat] Enable LTX-2 NVFP4 linear and attention QAT fine-tuning

### PR DESCRIPTION
**Pull request by SolitaryThinker**
## Purpose

Enable deployment-matched NVFP4 linear QAT and attention QAT for LTX-2.0 fine-tuning in the modular trainer. The recipe starts from `FastVideo/LTX2-Distilled-Diffusers` and builds on the LTX training support merged in #1624.

This is separate from #1619: it is a single-student `FineTuneMethod` run, not QAD/DMD2. It does not add LTX-2.3 QAT. The earlier validation-only NVFP4 FA4 integration has been removed; this PR now uses FastVideo's `ATTN_QAT_TRAIN` backend for training and `ATTN_QAT_INFER` for validation on supported sm120 hardware.

## Changes

- Resolve registered `pipeline.dit_config.quant_config` names during normal config loading and DCP-to-Diffusers raw-config reconstruction.
- Share the LTX-2 deployment target predicate with QAT. LTX-2.0 attaches QAT to the same 576 `LinearBase` modules used by NVFP4 deployment; existing Wan-style targeting is unchanged.
- Forward role-local `attention_backend` through `LTX2Model` and configure the recipe with `ATTN_QAT_TRAIN`.
- Allow QAT attention only on LTX-2's head-dim-128 video paths. Head-dim-64 audio stays dense, and masked text attention is explicitly pinned to `TORCH_SDPA`.
- Make unsupported role overrides honor a layer's explicit default backend, preventing a forced QAT backend from dropping the masked attention path.
- Add a validation-only `attn_qat_infer` context that swaps `AttnQatTrainImpl` to `AttnQatInferImpl`, restores it in `finally`, and refuses to claim availability outside sm120.
- Add an overfit recipe, selector/callback/wiring contracts, a full-model GB200 gradient regression, and the existing four-GPU nightly quality case.

## Hardware behavior

| Hardware | Training | Validation |
|---|---|---|
| GB200 / sm100 | `nvfp4_qat_train` linears + `ATTN_QAT_TRAIN` attention | Keep `ATTN_QAT_TRAIN`; `ATTN_QAT_INFER` is unavailable |
| RTX 5090 / sm120 | Full 18.9B QAT training does not fit one 32 GB GPU | `ATTN_QAT_INFER` is supported and verified |
| Multi-GPU sm120 | Same QAT training stack | Canonical recipe temporarily swaps to `ATTN_QAT_INFER` |

The recipe documents the GB200 override `--callbacks.validation.attn_qat_infer false`. This keeps observable validation enabled while avoiding a false claim that the sm120 kernel ran on GB200.

## Test Plan

```bash
pre-commit run --files <changed paths>

pytest -q \
  fastvideo/tests/attention/test_selector_role_override.py \
  fastvideo/tests/ops/quantization/test_nvfp4_ltx2_wiring.py \
  fastvideo/tests/train/callbacks/test_validation.py

CUDA_VISIBLE_DEVICES=0 pytest -q \
  'fastvideo/tests/train/methods/test_ltx2_finetune.py::test_ltx2_finetune_single_train_step[ltx2_nvfp4_qat]'

# RTX 5090, LTX-relevant non-causal head-dim-128 kernel case
pytest -q \
  'fastvideo-kernel/tests/test_attn_qat_infer.py::test_accuracy_sdpa[non_causal-1-4-384-128]' -s

# 4x GB200; retain validation on the train-time QAT backend
NUM_GPUS=4 bash examples/train/run.sh \
  examples/train/configs/overfit_ltx2_t2v_nvfp4_qat.yaml \
  --callbacks.validation.attn_qat_infer false
```

## Test Results

Exact PR head `6e2c88f2d7bc29377888924638bcb7811bd41a9b`:

- Scoped pre-commit: all hooks passed; GitHub's current-head pre-commit check also passed.
- GB200 selector, LTX NVFP4 wiring, and validation callback contracts: `40 passed`.
- Single GB200 full-model training step: passed in `33.30s` with 576 deployment-matched QAT linears, 96 configured `AttnQatTrainImpl` video paths, finite/nonzero gradients, and refreshed block-0 gradient norm `0.2421`.
- RTX 5090 `ATTN_QAT_INFER` correctness case: passed with cosine similarity `0.981758`; the actual FastVideo wrapper reported available, returned shape `(1, 384, 4, 128)`, and produced finite output. LTX uses the verified non-causal path.
- 4x GB200 overfit: completed `300/300` steps and all seven 8-step validations. Final W&B summary: loss `0.02476`, grad norm `0.06493`, steady-state step time `1.573s`. Videos and metrics are in [W&B run `n69njq6h`](https://wandb.ai/wlsaidhi/fastvideo_ltx2/runs/n69njq6h).

The GB200 run used `ATTN_QAT_TRAIN` for both training and validation because `ATTN_QAT_INFER` is sm120-only. The optimized inference backend was separately executed on the allocated RTX 5090; a combined live full-model run needs multi-GPU sm120 capacity.

## Checklist

- [x] Pre-commit passed on every changed path
- [x] Added or updated regression coverage
- [x] Updated training documentation and hardware constraints
- [x] Verified the full forward/backward on GB200
- [x] Verified `ATTN_QAT_INFER` on RTX 5090
- [x] Completed the 4x GB200 overfit with online W&B validation
- [x] Support matrix update is not needed; this enables training for an existing LTX-2.0 checkpoint

### CODE REVIEW

**Comment by SolitaryThinker** [comment]
Rebased onto latest main to pick up the LTX-2 persistent compile-cache fixes from #1602; the rebase itself has no content changes (range-diff clean).

**Comment by SolitaryThinker** [comment]
Rebased onto current main and added the missing GB200 nightly override for the SM120-only validation kernel; the original eight commits have no content changes (range-diff clean).

File: fastvideo/train/utils/config.py [hunk 1]
```diff
@@ -282,9 +282,28 @@ def _parse_pipeline_config(
...
+    This must live in ``_parse_pipeline_config`` (not ``load_run_config``)
+    because ``dcp_to_diffusers`` rebuilds models from a checkpoint's raw
+    config by calling ``_parse_pipeline_config`` directly.
+    """
+    dit_config = getattr(pipeline_config, "dit_config", None)
+    quant = getattr(dit_config, "quant_config", None)
+    if isinstance(quant, str):
+        from fastvideo.layers.quantization import (
+            get_quantization_config, )
+        dit_config.quant_config = get_quantization_config(quant)()
```
[the remark this answers is not in the document]
Comment (SolitaryThinker) [reply fastvideo/train/utils/config.py:304]: `getattr(None, "quant_config", None)` returns the supplied default rather than raising, so this line is already `None`-safe. Reproduced directly at the exact PR head; resolving.
